### PR TITLE
remove deprecated Exchange.getOut from camel-mail

### PR DESCRIPTION
**Pull request by zbendhiba**
removing deprecated getOut method on camel-mail component.
I did some cleaning in test classes, like removing the deprecated  Assert.asserThat

### CODE REVIEW

File: components/camel-mail/src/test/java/org/apache/camel/dataformat/mime/multipart/MimeMultipartDataFormatTest.java [hunk 1]
```diff
@@ -261,9 +257,10 @@ public void marhsalOnlyMixed() throws IOException {
         headers.put("X-AdditionalData", "additional data");
         addAttachment("application/octet-stream", "foobar", "attachment.bin", headers);
         Exchange result = template.send("direct:marshalonlymixed", exchange);
-        assertThat(result.getOut().getHeader("Content-Type", String.class), startsWith("multipart/mixed"));
-        String resultBody = result.getOut().getBody(String.class);
-        assertThat(resultBody, containsString("Content-Description: Sample Attachment Data"));
+        assertTrue(result.getMessage().getHeader("Content-Type", String.class).startsWith("multipart/mixed"));
+        //assertThat(result.getMessage().getHeader("Content-Type", String.class), startsWith("multipart/mixed"));
```
Comment (aldettinger) [inline components/camel-mail/src/test/java/org/apache/camel/dataformat/mime/multipart/MimeMultipartDataFormatTest.java:261]: I would remove this comment

Comment (zbendhiba) [reply components/camel-mail/src/test/java/org/apache/camel/dataformat/mime/multipart/MimeMultipartDataFormatTest.java:261]: yes. I forgot that one !!

Comment (zbendhiba) [reply components/camel-mail/src/test/java/org/apache/camel/dataformat/mime/multipart/MimeMultipartDataFormatTest.java]: done !!

Comment (aldettinger) [reply components/camel-mail/src/test/java/org/apache/camel/dataformat/mime/multipart/MimeMultipartDataFormatTest.java]: Well done @zbendhiba, it's rebased and pushed upstream now :)

**Pull request merged**